### ISSUE TITLE
Add missing parameters to the `__int__` method of certain tasks.

### ISSUE THREAD

**Issue by karthikrangasai** [PyTorchLightning/lightning-flash#729]
## 🚀 Feature

Add `optimizer_kwargs`, `scheduler`, and `scheduler_kwargs` parameters to the `__init__` method of all tasks that are missing it.

### Motivation

Many tasks don't have a way passing a learning rate scheduler or even different optimizer arguments to the optimizer of their choice.

### Pitch

Add 
```
        optimizer_kwargs: Optional[Dict[str, Any]] = None,
        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
        scheduler_kwargs: Optional[Dict[str, Any]] = None,
```
to the `__init__` methods of all tasks keeping the API consistent across all tasks. Update the documentation as well.

### Additional context

I have the fix ready for this to submit as a PR.

**Comment by ethanwharris** [comment]
Hi @karthikrangasai this sounds great, happy to have a PR 😃
